Microsoft.Toolkit.Uwp.UI.Animations/ConnectedAnimations/ConnectedAnimationHelper.cs

```diff
@@ -114,22 +121,28 @@ void loadedHandler(object s, RoutedEventArgs args)
                     {
                         if (props.IsListAnimation && parameter != null && ApiInformationHelper.IsCreatorsUpdateOrAbove)
                         {
-                            props.ListViewBase.ScrollIntoView(parameter);
-
-                            // give time to the UI thread to scroll the list
-                            var t = props.ListViewBase.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, async () =>
+                            foreach (var listAnimProperty in props.ListAnimProperties)
                             {
-                                try
+                                if (listAnimProperty.ListViewBase.ItemsSource is IEnumerable<object> items && items.Contains(parameter))
                                 {
-                                    var success = await props.ListViewBase.TryStartConnectedAnimationAsync(connectedAnimation, parameter, props.ElementName);
+                                    listAnimProperty.ListViewBase.ScrollIntoView(parameter);
+
+                                    // give time to the UI thread to scroll the list
+                                    var t = listAnimProperty.ListViewBase.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, async () =>
```

Comment:
just out of curiosity what if props.ListAnimProperties has 10 items ? we will create 10 tasks to do the job ? isn't overkilling ?

Reply:
That would happen if the page has multiple lists that all have the same element (being passed as parameter in the page navigation) and the developers is trying to run the animation on every single list. We could prevent that by exiting the foreach loop after the first list, but I don't think it's necessary

Reply:
I reckon then we should mention that.